src/components/Posts/AllPost/AllPost.js

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect } from 'react';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Paper from '@material-ui/core/Paper';
+import Typography from '@material-ui/core/Typography';
+import Grid from '@material-ui/core/Grid';
+import Card from '@material-ui/core/Card';
+import CardActionArea from '@material-ui/core/CardActionArea';
+import CardContent from '@material-ui/core/CardContent';
+import CardMedia from '@material-ui/core/CardMedia';
+import Hidden from '@material-ui/core/Hidden';
+import Button from '@material-ui/core/Button';
+import Container from '@material-ui/core/Container';
+import Chip from '@material-ui/core/Chip';
+import cookie from 'react-cookies';
+import { Link } from 'react-router-dom';
+import { serviceRequest } from '../../../services/serviceRequest';
+import config from '../../../config';
+import { StateProvider } from '../../StateProvider';
+import useStyles from './style';
+
+const path = '/api/v1/user/getposts?';
+const domain = config.apiDomain;
+
+function preparePayload(method, headers, params) {
+  const url = `https://${domain}${path}`;
+  return {
+    method,
+    url,
+    headers,
+    params,
+  };
+}
+
+export const AllPost = () => {
+  const classes = useStyles();
+  const allPostParams = { page: 1, pageSize: 5, viewall: true };
+  const token = cookie.load('jwtToken');
+  const allPostHeaders = { Authorization: token };
+  const [allPosts, setAllPosts] = useState({ posts: [] });
+  const main = allPosts.posts[0];
+  const subPosts = allPosts.posts.slice(1);
+
+  const initialState = {
+    page: { offset: 0, limit: 1 },
+  };
+
+  const reducer = (state, action) => {
+    switch (action.type) {
+      case 'changePage':
+        return {
+          ...state,
+          page: action.newPage,
+        };
+
+      default:
+        return state;
+    }
+  };
+
+  useEffect(() => {
+    async function fetchAllPosts() {
+      const requestPayload = preparePayload('get', allPostHeaders, allPostParams);
+      const response = await serviceRequest(requestPayload);
+      try {
+        if (response.status && response.status === 'success') {
+          setAllPosts(response.data);
+        } else if (response.status && response.status === 'error') {
+          throw new Error('Authorization Error');
+        } else {
+          throw new Error('Internal Service Error');
+        }
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      }
+    }
+    fetchAllPosts();
+  }, [initialState.page.offset, initialState.page.limit, allPostHeaders, allPostParams]);
+
+  return (
+    <StateProvider initialState={initialState} reducer={reducer}>
+      <CssBaseline />
+      <Container maxWidth="lg">
+        <main>
+          <Paper className={classes.mainFeaturedPost}>
+            {
+              <img
+                style={{ display: 'none' }}
+                src="https://source.unsplash.com/user/erondu"
```

Comment:
should download the image and optimize it

Reply:
We are not sure if we will be allowing user to add image to their post.  Links from unsplash are stable for us to use. Another reason I want to use them is because these links provided random image each time you make a call to the link.